src/wrapper/java_vm/vm.rs

```diff
@@ -42,9 +41,10 @@ use InitArgs;
 /// and [auto locals](struct.JNIEnv.html#method.auto_local) in loops.
 ///
 /// Jni-rs provides a helper struct to deal with such local references -
-/// [`Executor`](struct.Executor.html). It is a simple abstraction that hides the necessity
-/// of handling local references behind the
-/// [`Executor::with_attached`](struct.Executor.html#method.with_attached) method.
+/// [`Executor`](struct.Executor.html). The `Executor` provides `JNIEnv` to any native thread with
+/// reasonably high performance, making use of permanent attaches. It also removes the risk of local
+/// references leaks if used consistently. `Executor` is recommended to use when the scope of
+/// necessary attachment is distinct.
```

Comment:
What do you mean by 'distinct' scope?

Reply:
When the scope is relatively small and well defined (e.g. we're not using JNIEnv during the whole program execution). Perhaps it's not the best wording here

Reply:
I don't think 'distinct' works here. I also don't understand why the scope has to be 'relatively small and well defined' — Exonum uses it in an unbounded scope.

Reply:
The scope of attachment is always restricted by closure passed into `with_attached` method, I mean this. Scope is `unbounded` when we attach the main thread and do not detach it to the end of the program execution, this is my view of the definition.

Reply:
Well, there is a scope of thread attachment — where the native thread remains attached, and a scope in which the JNI interface pointer is accessible. The first may be unbounded for Executors, the second is limited.